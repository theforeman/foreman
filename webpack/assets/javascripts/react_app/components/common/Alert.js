import React from 'react';
import Icon from './Icon';
import { ALERT_CSS } from '../../constants';

const Alert = (props) => {
  let classNames = ALERT_CSS[props.type];

  if (props.css) {
    classNames += ' ' + props.css;
  }

  if (props.dismissable) {
    classNames += ' ' + ALERT_CSS.dismissable;
  }

  const closeBtn = (
    <button className="close" data-dismiss="alert" aria-hidden="true">
      <Icon type="close" />
    </button>
  );

  return (
    <div className={classNames}>
      {props.dismissable && closeBtn}
      {props.children}
    </div>
  );
};

Alert.defaultProps = { dismissable: false, css: null, type: 'info' };
export default Alert;
