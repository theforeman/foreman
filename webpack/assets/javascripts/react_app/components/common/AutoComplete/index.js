import React, { Component } from 'react';
import Downshift from 'downshift';
import { Dropdown, MenuItem, InputGroup, FormControl, Button } from 'patternfly-react';
import PropTypes from 'prop-types';

export const renderItems = ({
  items, highlightedIndex, selectedItem, getItemProps,
}) => (
  <Dropdown.Menu
    style={{
      display: 'block',
      position: 'absolute',
      left: 'inherit',
      marginTop: 0,
      top: 'auto',
    }}
  >
    {items.map((item, index) => (
      <MenuItem
        {...getItemProps({
          index,
          item,
          active: highlightedIndex === index,
          onClick: (e) => {
            // At this point the event.defaultPrevented
            // is already set to true by react-bootstrap
            // MenuItem. We need to set it back to false
            // So downshift will execute it's own handler
            e.defaultPrevented = false;
          },
        })}
        key={item}
      >
        {item}
      </MenuItem>
    ))}
  </Dropdown.Menu>
);

class AutoComplete extends Component {
  constructor(props) {
    super(props);

    this.state = {
      inputValue: '',
    };
  }

  render() {
    const {
      onSearch, labelText, onInputUpdate, items, actionText, ...rest
    } = this.props;

    return (
      <Downshift
        onStateChange={({ inputValue }) => {
          if (typeof inputValue === 'string') {
            onInputUpdate(inputValue);
            this.setState({ inputValue });
          }
        }}
        defaultHighlightedIndex={0}
        selectedItem={this.state.inputValue}
        {...rest}
      >
        {({
          getInputProps,
          getItemProps,
          getLabelProps,
          isOpen,
          inputValue,
          highlightedIndex,
          getRootProps,
          selectedItem,
        }) => (
          <div>
            {labelText && <label {...getLabelProps()}>{labelText}</label>}
            <InputGroup>
              <FormControl type="text" {...getInputProps()} />
              <InputGroup.Button>
                <Button onClick={() => onSearch(inputValue)}>{actionText}</Button>
              </InputGroup.Button>
            </InputGroup>

            {isOpen && items.length
              ? renderItems({
                  items,
                  highlightedIndex,
                  selectedItem,
                  getItemProps,
                })
              : null}
          </div>
        )}
      </Downshift>
    );
  }
}

AutoComplete.propTypes = {
  items: PropTypes.arrayOf(PropTypes.string).isRequired,
  onInputUpdate: PropTypes.func.isRequired,
  onSearch: PropTypes.func.isRequired,
  labelText: PropTypes.string,
  actionText: PropTypes.string,
};

AutoComplete.defaultProps = {
  labelText: '',
  actionText: 'Search',
};

export default AutoComplete;
