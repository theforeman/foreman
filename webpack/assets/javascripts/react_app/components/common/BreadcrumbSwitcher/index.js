import React from 'react';
import {
  Popover,
  Button,
  Icon,
  ListGroup,
  ListGroupItem,
  OverlayTrigger,
} from 'patternfly-react';
import { ButtonToolbar } from 'react-bootstrap';
import EllipsisWithTooltip from 'react-ellipsis-with-tooltip';
import PropTypes from 'prop-types';
import './BreadcrumbSwitcher.scss';

const BreadcrumbSwitcher = ({ resources }) => {
  const popover = (
    <Popover id="popover-contained">
      <ListGroup className="scrollable-list">
        {resources.map((item) => {
          const { onClick, url, caption } = item;
          return (
            <ListGroupItem
              className="no-border"
              key={`id-${caption}`}
              href={url}
              onClick={onClick}
            >
              <EllipsisWithTooltip>{caption}</EllipsisWithTooltip>
            </ListGroupItem>
          );
        })}
      </ListGroup>
    </Popover>
  );
  return (
    <ButtonToolbar>
      <OverlayTrigger
        rootClose
        trigger="click"
        placement="bottom"
        overlay={popover}
      >
        <Button>
          <Icon type="fa" name="exchange" />
        </Button>
      </OverlayTrigger>
    </ButtonToolbar>
  );
};

BreadcrumbSwitcher.propTypes = {
  href: PropTypes.string,
  onClick: PropTypes.func,
  caption: PropTypes.string,
};

export default BreadcrumbSwitcher;
