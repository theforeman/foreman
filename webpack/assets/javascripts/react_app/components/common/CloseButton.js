import React from 'react';
import Icon from './Icon';

const CloseButton = ({onClick}) => {
  return (
    <button className="close" aria-hidden="true" onClick={onClick}>
      <Icon type="close" />
    </button>
  );
};

CloseButton.propTypes = {
  onClick: React.PropTypes.func.isRequired
};

export default CloseButton;
