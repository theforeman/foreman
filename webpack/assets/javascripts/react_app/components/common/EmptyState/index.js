import EmptyStatePattern from './EmptyStatePattern';
import DefaultEmptyState from './DefaultEmptyState';

export default DefaultEmptyState;
export { EmptyStatePattern };
