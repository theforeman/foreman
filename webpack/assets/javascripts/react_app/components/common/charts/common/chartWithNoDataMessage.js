import React from 'react';
import MessageBox from '../../MessageBox';

export const chartWithNoDataMessage = (WrappedComp, noDataMsg = 'No data available') => (props) => {
  const columns = props && props.data && props.data.columns ? props.data.columns : [];
  if (columns.length > 0) {
    return (<WrappedComp {...props} />);
  }
  return <MessageBox msg={noDataMsg} icontype="info" />;
};
