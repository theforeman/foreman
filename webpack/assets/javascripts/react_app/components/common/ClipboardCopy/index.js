import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Button, OverlayTrigger, Tooltip } from 'patternfly-react';
import UUID from 'uuid/v1';
import { translate as __ } from '../../../common/I18n';
import './clipboard-copy.scss';

const ClipboardCopy = ({
  text: defaultText,
  successMessage,
  buttonText,
  textareaProps,
  buttonProps,
}) => {
  const [text, setText] = useState(defaultText);

  return (
    <div className="clipboard-copy">
      <textarea
        defaultValue={text}
        onChange={({ target: { value } }) => setText(value)}
        {...textareaProps}
      />
      <OverlayTrigger
        overlay={<Tooltip id={UUID()}>{successMessage}</Tooltip>}
        placement="right"
        trigger={['click']}
        rootClose
      >
        <Button
          onClick={() => navigator.clipboard.writeText(text)}
          bsStyle="default"
          {...buttonProps}
        >
          {buttonText}
        </Button>
      </OverlayTrigger>
    </div>
  );
};

ClipboardCopy.propTypes = {
  text: PropTypes.string.isRequired,
  buttonText: PropTypes.string,
  successMessage: PropTypes.string,
  textareaProps: PropTypes.object,
  buttonProps: PropTypes.object,
};

ClipboardCopy.defaultProps = {
  buttonText: __('Copy to clipboard'),
  successMessage: __('Copied!'),
  textareaProps: {},
  buttonProps: {},
};

export default ClipboardCopy;
