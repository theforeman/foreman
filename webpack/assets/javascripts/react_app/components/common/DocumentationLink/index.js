import React from 'react';
import PropTypes from 'prop-types';
import { MenuItem, Icon } from 'patternfly-react';
import { newWindowOnClick } from '../../../common/helpers';
import { translate as __ } from '../../../../react_app/common/I18n';

const DocumentationLink = ({ href, children }) => (
  <MenuItem key="documentationUrl" href={href} onClick={newWindowOnClick(href)}>
    <Icon type="fa" name="question-circle" />
    {` ${children}`}
  </MenuItem>
);

DocumentationLink.propTypes = {
  href: PropTypes.string.isRequired,
  children: PropTypes.node,
};

DocumentationLink.defaultProps = {
  children: __('Documentation'),
};

export default DocumentationLink;
