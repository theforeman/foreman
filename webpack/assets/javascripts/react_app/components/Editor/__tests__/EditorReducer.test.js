import {
  EDITOR_CHANGE_DIFF_VIEW,
  EDITOR_CHANGE_SETTING,
  EDITOR_CHANGE_TAB,
  EDITOR_CHANGE_VALUE,
  EDITOR_DISMISS_ERROR,
  EDITOR_SHOW_ERROR,
  EDITOR_EXEC_PREVIEW,
  EDITOR_IMPORT_FILE,
  EDITOR_INITIALIZE,
  EDITOR_MODAL_TOGGLE,
  EDITOR_REVERT_CHANGES,
  EDITOR_TOGGLE_MASK,
  EDITOR_TOGGLE_RENDER_VIEW,
  EDITOR_HOST_SELECT_CLEAR,
  EDITOR_HOST_SELECT_TOGGLE,
  EDITOR_FETCH_HOST_PENDING,
  EDITOR_HOST_SELECT_RESET,
  EDITOR_FETCH_HOST_RESOLVED,
  EDITOR_HOST_INITIAL_FETCH,
  EDITOR_SHOW_LOADING,
  EDITOR_HIDE_LOADING,
} from '../EditorConstants';

import reducer from '../EditorReducer';

import { testReducerSnapshotWithFixtures } from '../../../common/testHelpers';

const fixtures = {
  'should return the initial state': {},
  'should handle EDITOR_INITIALIZE': {
    action: {
      type: EDITOR_INITIALIZE,
      payload: { value: 'newValue' },
    },
  },
  'should handle EDITOR_CHANGE_STATE': {
    action: {
      type: EDITOR_CHANGE_VALUE,
      payload: '< newValue />',
    },
  },
  'should handle EDITOR_IMPORT_FILE': {
    action: {
      type: EDITOR_IMPORT_FILE,
      payload: {
        value: '</>',
      },
    },
  },
  'should handle EDITOR_REVERT_CHANGES': {
    action: {
      type: EDITOR_REVERT_CHANGES,
      payload: {
        value: '</>',
      },
    },
  },
  'should handle EDITOR_EXEC_PREVIEW': {
    action: {
      type: EDITOR_EXEC_PREVIEW,
      payload: {
        previewResult: '</>',
      },
    },
  },
  'should handle EDITOR_MODAL_TOGGLE': {
    action: {
      type: EDITOR_MODAL_TOGGLE,
    },
  },
  'should handle EDITOR_CHANGE_DIFF_VIEW': {
    action: {
      type: EDITOR_CHANGE_DIFF_VIEW,
      payload: 'unified',
    },
  },
  'should handle EDITOR_SHOW_ERROR': {
    action: {
      type: EDITOR_SHOW_ERROR,
      payload: {
        showError: true,
        errorText: 'error',
        previewResult: 'error',
      },
    },
  },
  'should handle EDITOR_DISMISS_ERROR': {
    action: {
      type: EDITOR_DISMISS_ERROR,
      payload: { showError: false, errorText: '' },
    },
  },
  'should handle EDITOR_CHANGE_TAB': {
    action: {
      type: EDITOR_CHANGE_TAB,
      payload: 'diff',
    },
  },
  'should handle EDITOR_TOGGLE_MASK': {
    action: {
      type: EDITOR_TOGGLE_MASK,
    },
  },
  'should handle EDITOR_TOGGLE_RENDER_VIEW': {
    action: {
      type: EDITOR_TOGGLE_RENDER_VIEW,
    },
  },
  'should handle EDITOR_HOST_SELECT_CLEAR': {
    action: {
      type: EDITOR_HOST_SELECT_CLEAR,
    },
  },
  'should handle EDITOR_CHANGE_SETTING': {
    action: {
      type: EDITOR_CHANGE_SETTING,
      payload: { mode: 'html' },
    },
  },
  'should handle EDITOR_SHOW_LOADING': {
    action: {
      type: EDITOR_SHOW_LOADING,
    },
  },
  'should handle EDITOR_HIDE_LOADING': {
    action: {
      type: EDITOR_HIDE_LOADING,
    },
  },
  'should handle EDITOR_HOST_INITIAL_FETCH': {
    action: {
      type: EDITOR_HOST_INITIAL_FETCH,
      payload: [{ id: '1', name: 'html' }],
    },
  },
  'should handle EDITOR_HOST_SELECT_TOGGLE': {
    action: {
      type: EDITOR_HOST_SELECT_TOGGLE,
    },
  },
  'should handle EDITOR_HOST_SELECT_RESET': {
    action: {
      type: EDITOR_HOST_SELECT_RESET,
    },
  },
  'should handle EDITOR_FETCH_HOST_RESOLVED': {
    action: {
      type: EDITOR_FETCH_HOST_RESOLVED,
      payload: {
        hosts: [{ id: '1', name: 'html' }],
      },
    },
  },
  'should handle EDITOR_FETCH_HOST_PENDING': {
    action: {
      type: EDITOR_FETCH_HOST_PENDING,
      payload: {
        isFetchingHosts: true,
        searchQuery: 'value',
        isSearchingHosts: true,
      },
    },
  },
};
describe('Editor reducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
