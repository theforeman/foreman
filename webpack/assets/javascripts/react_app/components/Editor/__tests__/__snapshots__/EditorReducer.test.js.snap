// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor reducer should handle EDITOR_CHANGE_DIFF_VIEW 1`] = `
Object {
  "diffViewType": "unified",
  "editorName": "editor",
  "errorText": "",
  "filteredHosts": Array [],
  "hosts": Array [],
  "isFetchingHosts": false,
  "isLoading": false,
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "isSearchingHosts": false,
  "isSelectOpen": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "previewResult": "",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "input",
  "showError": false,
  "templateClass": "",
  "theme": "Monokai",
  "value": "",
}
`;

exports[`Editor reducer should handle EDITOR_CHANGE_SETTING 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "filteredHosts": Array [],
  "hosts": Array [],
  "isFetchingHosts": false,
  "isLoading": false,
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "isSearchingHosts": false,
  "isSelectOpen": false,
  "keyBinding": "Default",
  "mode": "html",
  "previewResult": "",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "input",
  "showError": false,
  "templateClass": "",
  "theme": "Monokai",
  "value": "",
}
`;

exports[`Editor reducer should handle EDITOR_CHANGE_STATE 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "filteredHosts": Array [],
  "hosts": Array [],
  "isFetchingHosts": false,
  "isLoading": false,
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "isSearchingHosts": false,
  "isSelectOpen": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "previewResult": "",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "input",
  "showError": false,
  "templateClass": "",
  "theme": "Monokai",
  "value": "< newValue />",
}
`;

exports[`Editor reducer should handle EDITOR_CHANGE_TAB 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "filteredHosts": Array [],
  "hosts": Array [],
  "isFetchingHosts": false,
  "isLoading": false,
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "isSearchingHosts": false,
  "isSelectOpen": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "previewResult": "",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "diff",
  "showError": false,
  "templateClass": "",
  "theme": "Monokai",
  "value": "",
}
`;

exports[`Editor reducer should handle EDITOR_DISMISS_ERROR 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "filteredHosts": Array [],
  "hosts": Array [],
  "isFetchingHosts": false,
  "isLoading": false,
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "isSearchingHosts": false,
  "isSelectOpen": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "previewResult": "",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "input",
  "showError": false,
  "templateClass": "",
  "theme": "Monokai",
  "value": "",
}
`;

exports[`Editor reducer should handle EDITOR_EXEC_PREVIEW 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "filteredHosts": Array [],
  "hosts": Array [],
  "isFetchingHosts": false,
  "isLoading": false,
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "isSearchingHosts": false,
  "isSelectOpen": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "previewResult": "</>",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "input",
  "showError": false,
  "templateClass": "",
  "theme": "Monokai",
  "value": "",
}
`;

exports[`Editor reducer should handle EDITOR_FETCH_HOST_PENDING 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "filteredHosts": Array [],
  "hosts": Array [],
  "isFetchingHosts": true,
  "isLoading": false,
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "isSearchingHosts": true,
  "isSelectOpen": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "previewResult": "",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "value",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "input",
  "showError": false,
  "templateClass": "",
  "theme": "Monokai",
  "value": "",
}
`;

exports[`Editor reducer should handle EDITOR_FETCH_HOST_RESOLVED 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "filteredHosts": Array [],
  "hosts": Array [
    Object {
      "id": "1",
      "name": "html",
    },
  ],
  "isFetchingHosts": false,
  "isLoading": false,
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "isSearchingHosts": false,
  "isSelectOpen": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "previewResult": "",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "input",
  "showError": false,
  "templateClass": "",
  "theme": "Monokai",
  "value": "",
}
`;

exports[`Editor reducer should handle EDITOR_HIDE_LOADING 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "filteredHosts": Array [],
  "hosts": Array [],
  "isFetchingHosts": false,
  "isLoading": false,
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "isSearchingHosts": false,
  "isSelectOpen": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "previewResult": "",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "input",
  "showError": false,
  "templateClass": "",
  "theme": "Monokai",
  "value": "",
}
`;

exports[`Editor reducer should handle EDITOR_HOST_INITIAL_FETCH 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "filteredHosts": Array [],
  "hosts": Array [
    Object {
      "id": "1",
      "name": "html",
    },
  ],
  "isFetchingHosts": false,
  "isLoading": false,
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "isSearchingHosts": false,
  "isSelectOpen": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "previewResult": "",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "input",
  "showError": false,
  "templateClass": "",
  "theme": "Monokai",
  "value": "",
}
`;

exports[`Editor reducer should handle EDITOR_HOST_SELECT_CLEAR 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "filteredHosts": Array [],
  "hosts": Array [],
  "isFetchingHosts": false,
  "isLoading": false,
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "isSearchingHosts": false,
  "isSelectOpen": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "previewResult": "",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "input",
  "showError": false,
  "templateClass": "",
  "theme": "Monokai",
  "value": "",
}
`;

exports[`Editor reducer should handle EDITOR_HOST_SELECT_RESET 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "filteredHosts": Array [],
  "hosts": Array [],
  "isFetchingHosts": false,
  "isLoading": false,
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "isSearchingHosts": false,
  "isSelectOpen": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "previewResult": "",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "input",
  "showError": false,
  "templateClass": "",
  "theme": "Monokai",
  "value": "",
}
`;

exports[`Editor reducer should handle EDITOR_HOST_SELECT_TOGGLE 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "filteredHosts": Array [],
  "hosts": Array [],
  "isFetchingHosts": false,
  "isLoading": false,
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "isSearchingHosts": false,
  "isSelectOpen": true,
  "keyBinding": "Default",
  "mode": "Ruby",
  "previewResult": "",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "input",
  "showError": false,
  "templateClass": "",
  "theme": "Monokai",
  "value": "",
}
`;

exports[`Editor reducer should handle EDITOR_IMPORT_FILE 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "filteredHosts": Array [],
  "hosts": Array [],
  "isFetchingHosts": false,
  "isLoading": false,
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "isSearchingHosts": false,
  "isSelectOpen": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "previewResult": "",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "input",
  "showError": false,
  "templateClass": "",
  "theme": "Monokai",
  "value": "</>",
}
`;

exports[`Editor reducer should handle EDITOR_INITIALIZE 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "filteredHosts": Array [],
  "hosts": Array [],
  "isFetchingHosts": false,
  "isLoading": false,
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "isSearchingHosts": false,
  "isSelectOpen": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "previewResult": "",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "input",
  "showError": false,
  "templateClass": "",
  "theme": "Monokai",
  "value": "newValue",
}
`;

exports[`Editor reducer should handle EDITOR_MODAL_TOGGLE 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "filteredHosts": Array [],
  "hosts": Array [],
  "isFetchingHosts": false,
  "isLoading": false,
  "isMasked": false,
  "isMaximized": true,
  "isRendering": false,
  "isSearchingHosts": false,
  "isSelectOpen": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "previewResult": "",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "input",
  "showError": false,
  "templateClass": "",
  "theme": "Monokai",
  "value": "",
}
`;

exports[`Editor reducer should handle EDITOR_REVERT_CHANGES 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "filteredHosts": Array [],
  "hosts": Array [],
  "isFetchingHosts": false,
  "isLoading": false,
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "isSearchingHosts": false,
  "isSelectOpen": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "previewResult": "",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "input",
  "showError": false,
  "templateClass": "",
  "theme": "Monokai",
  "value": "</>",
}
`;

exports[`Editor reducer should handle EDITOR_SHOW_ERROR 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "error",
  "filteredHosts": Array [],
  "hosts": Array [],
  "isFetchingHosts": false,
  "isLoading": false,
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "isSearchingHosts": false,
  "isSelectOpen": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "previewResult": "error",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "input",
  "showError": true,
  "templateClass": "",
  "theme": "Monokai",
  "value": "",
}
`;

exports[`Editor reducer should handle EDITOR_SHOW_LOADING 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "filteredHosts": Array [],
  "hosts": Array [],
  "isFetchingHosts": false,
  "isLoading": true,
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "isSearchingHosts": false,
  "isSelectOpen": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "previewResult": "",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "input",
  "showError": false,
  "templateClass": "",
  "theme": "Monokai",
  "value": "",
}
`;

exports[`Editor reducer should handle EDITOR_TOGGLE_MASK 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "filteredHosts": Array [],
  "hosts": Array [],
  "isFetchingHosts": false,
  "isLoading": false,
  "isMasked": true,
  "isMaximized": false,
  "isRendering": false,
  "isSearchingHosts": false,
  "isSelectOpen": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "previewResult": "",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "input",
  "showError": false,
  "templateClass": "",
  "theme": "Monokai",
  "value": "",
}
`;

exports[`Editor reducer should handle EDITOR_TOGGLE_RENDER_VIEW 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "filteredHosts": Array [],
  "hosts": Array [],
  "isFetchingHosts": false,
  "isLoading": false,
  "isMasked": false,
  "isMaximized": false,
  "isRendering": true,
  "isSearchingHosts": false,
  "isSelectOpen": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "previewResult": "",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "input",
  "showError": false,
  "templateClass": "",
  "theme": "Monokai",
  "value": "",
}
`;

exports[`Editor reducer should return the initial state 1`] = `
Object {
  "diffViewType": "split",
  "editorName": "editor",
  "errorText": "",
  "filteredHosts": Array [],
  "hosts": Array [],
  "isFetchingHosts": false,
  "isLoading": false,
  "isMasked": false,
  "isMaximized": false,
  "isRendering": false,
  "isSearchingHosts": false,
  "isSelectOpen": false,
  "keyBinding": "Default",
  "mode": "Ruby",
  "previewResult": "",
  "readOnly": false,
  "renderedEditorValue": "",
  "searchQuery": "",
  "selectedHost": Object {
    "id": "",
    "name": "",
  },
  "selectedView": "input",
  "showError": false,
  "templateClass": "",
  "theme": "Monokai",
  "value": "",
}
`;
